src/Elders.Cronus/Hosting/CronusHost.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+
+namespace Elders.Cronus
+{
+    public sealed class CronusHost : ICronusHost
+    {
+        private readonly IConsumer<IAggregateRootApplicationService> appServices;
+        private readonly IConsumer<IProjection> projections;
+        private readonly IConsumer<IPort> ports;
+        private readonly IConsumer<ISaga> sagas;
+        private readonly IConsumer<IGateway> gateways;
+        private readonly ProjectionsBootstrapper projectionsBootstrapper;
+
+        public CronusHost(IConsumer<IAggregateRootApplicationService> appServices, IConsumer<IProjection> projections, IConsumer<IPort> ports, IConsumer<ISaga> sagas, IConsumer<IGateway> gateways, ProjectionsBootstrapper projectionsBootstrapper)
+        {
+            this.appServices = appServices ?? throw new ArgumentNullException(nameof(appServices));
+            this.projections = projections ?? throw new ArgumentNullException(nameof(appServices));
+            this.ports = ports ?? throw new ArgumentNullException(nameof(appServices));
+            this.sagas = sagas ?? throw new ArgumentNullException(nameof(appServices));
+            this.gateways = gateways ?? throw new ArgumentNullException(nameof(appServices));
+            this.projectionsBootstrapper = projectionsBootstrapper;
+        }
+
+        public void Start()
+        {
+            appServices.Start();
+            projections.Start();
+            ports.Start();
+            sagas.Start();
+            gateways.Start();
+
+            projectionsBootstrapper.Bootstrap();
+        }
+
+        public void Stop()
```

Comment:
If any of the services fails to stop, the others will never be stopped.

Reply:
Ok, what do you suggest?

Reply:
try-catch and/or logs

Reply:
Dunni, @Elders/cronus for more suggestions